2010-01-25  Yaakov Selkowitz  <yselkowitz@users.sourceforge.net>

	* configure.ac (EXTENSION_LIBTOOL_FLAGS) [platform_win32]: update location
	of gthumb executable import library.
	* extensions/*/Makefile.am: add extra *_LIBS as necessary.
	* extensions/jpeg_utils/transupp.c: Remove obsolete Cygwin exception.

--- origsrc/gthumb-2.11.1/configure.ac	2010-01-04 10:33:06.000000000 -0600
+++ src/gthumb-2.11.1/configure.ac	2010-01-25 15:56:01.051008600 -0600
@@ -204,7 +204,7 @@ AC_MSG_RESULT([$os_win32])
 AM_CONDITIONAL(OS_WIN32, test "$os_win32" = "yes")
 
 if test "$platform_win32" = "yes" ; then
-	EXTENSION_LIBTOOL_FLAGS='-module -avoid-version -no-undefined -Wl,$(top_builddir)/src/.libs/gthumb.exe.a'
+	EXTENSION_LIBTOOL_FLAGS='-module -avoid-version -no-undefined -Wl,$(top_builddir)/gthumb/.libs/gthumb.exe.a'
 else
 	EXTENSION_LIBTOOL_FLAGS='-module -avoid-version -disable-static'
 fi
--- origsrc/gthumb-2.11.1/extensions/exiv2/Makefile.am	2010-01-04 10:33:06.000000000 -0600
+++ src/gthumb-2.11.1/extensions/exiv2/Makefile.am	2010-01-25 16:16:34.492557400 -0600
@@ -16,7 +16,7 @@ libexiv2_la_SOURCES = 			\
 
 libexiv2_la_CPPFLAGS = $(GTHUMB_CFLAGS) $(EXIV2_CFLAGS) -I$(top_srcdir) -I$(top_builddir)/gthumb 
 libexiv2_la_LDFLAGS = $(EXTENSION_LIBTOOL_FLAGS)
-libexiv2_la_LIBADD = $(GTHUMB_LIBS) 
+libexiv2_la_LIBADD = $(GTHUMB_LIBS) $(EXIV2_LIBS)
 libexiv2_la_DEPENDENCIES = $(top_builddir)/gthumb/gthumb$(EXEEXT)
 
 extensioninidir = $(extensiondir)
--- origsrc/gthumb-2.11.1/extensions/gstreamer/Makefile.am	2010-01-04 10:33:06.000000000 -0600
+++ src/gthumb-2.11.1/extensions/gstreamer/Makefile.am	2010-01-25 16:35:57.566081400 -0600
@@ -21,7 +21,7 @@ libgstreamer_la_SOURCES = 			\
 
 libgstreamer_la_CPPFLAGS = $(GTHUMB_CFLAGS) $(GSTREAMER_CFLAGS) -I$(top_srcdir) -I$(top_builddir)/gthumb 
 libgstreamer_la_LDFLAGS = $(EXTENSION_LIBTOOL_FLAGS)
-libgstreamer_la_LIBADD = $(GTHUMB_LIBS) 
+libgstreamer_la_LIBADD = $(GTHUMB_LIBS) $(GSTREAMER_LIBS)
 libgstreamer_la_DEPENDENCIES = $(top_builddir)/gthumb/gthumb$(EXEEXT)
 
 extensioninidir = $(extensiondir)
--- origsrc/gthumb-2.11.1/extensions/jpeg_utils/transupp.c	2010-01-04 10:33:06.000000000 -0600
+++ src/gthumb-2.11.1/extensions/jpeg_utils/transupp.c	2010-01-25 18:44:39.797767800 -0600
@@ -49,11 +49,6 @@
 #define MAX(a, b)  (((a) > (b)) ? (a) : (b))
 #endif
 
-#ifdef __CYGWIN__
-#define height_in_blocks height_in_data_units
-#define width_in_blocks  width_in_data_units
-#endif
-
 enum {
   JERR_CONVERSION_NOTIMPL
 };
--- origsrc/gthumb-2.11.1/extensions/raw_files/Makefile.am	2010-01-04 10:33:06.000000000 -0600
+++ src/gthumb-2.11.1/extensions/raw_files/Makefile.am	2010-01-25 18:18:05.188561300 -0600
@@ -6,7 +6,7 @@ libraw_files_la_SOURCES = \
 
 libraw_files_la_CFLAGS = $(GTHUMB_CFLAGS) $(LIBOPENRAW_CFLAGS) -I$(top_srcdir) -I$(top_builddir)/gthumb 
 libraw_files_la_LDFLAGS = $(EXTENSION_LIBTOOL_FLAGS)
-libraw_files_la_LIBADD = $(GTHUMB_LIBS)
+libraw_files_la_LIBADD = $(GTHUMB_LIBS) $(LIBOPENRAW_LIBS)
 libraw_files_la_DEPENDENCIES = $(top_builddir)/gthumb/gthumb$(EXEEXT)
 
 extensioninidir = $(extensiondir)
--- origsrc/gthumb-2.11.1/extensions/slideshow/Makefile.am	2010-01-04 10:33:06.000000000 -0600
+++ src/gthumb-2.11.1/extensions/slideshow/Makefile.am	2010-01-25 18:35:09.000120000 -0600
@@ -20,7 +20,7 @@ libslideshow_la_SOURCES = 		\
 
 libslideshow_la_CFLAGS = $(GTHUMB_CFLAGS) $(CLUTTER_CFLAGS) -I$(top_srcdir) -I$(top_builddir)/gthumb 
 libslideshow_la_LDFLAGS = $(EXTENSION_LIBTOOL_FLAGS)
-libslideshow_la_LIBADD = $(GTHUMB_LIBS)
+libslideshow_la_LIBADD = $(GTHUMB_LIBS) $(CLUTTER_LIBS)
 libslideshow_la_DEPENDENCIES = $(top_builddir)/gthumb/gthumb$(EXEEXT)
 
 extensioninidir = $(extensiondir)
